### DISCUSSION TITLE
Svg node

### DISCUSSION THREAD

**Discussion by AceroMagico** [WASasquatch/was-node-suite-comfyui#20] [Ideas]
I am very happy with the success of your suite, you deserve it.

Idea for new nodes:

Svg importer (convert svg to image with the desired resolution) (may this is good: https://pypi.org/project/svglib/)
This allows you to use a myriad of ready-made free svg as a source and for your original images to create, for example in affinity Inkscape Illustrator etc, to have only a single master independent of the resolution you then want to use in ComfyUI.

Also by having the Svgs Conversion Node you can integrate a whole range of new effects that are contained in PlotterFun :

https://mitxela.com/plotterfun/
https://github.com/mitxela/plotterfun
https://www.opensourceagenda.com/projects/plotterfun

these in combination with ControlNet open up endless possibilities for creating.

See too this for the reverse (image to Svg):  (use Potrace https://potrace.sourceforge.net/)
https://github.com/GeorgLegato/Txt2Vectorgraphics
https://github.com/GeorgLegato/stable-diffusion-webui-vectorstudio

**Comment by WASasquatch** [comment]
I thought I responded to this, but apparently not. I apologize x.x

A SVG importer is a great idea, and I have added that to my list. 

However, ComfyUI currently doesn't have arbitrary support for mixing in other GUIs, let alone even the ability to output simple text to a node after it's been ran. 

I also don't see the reason to add SVG editors, when there already is a raster image editor nodes that do the same things -- and you'd be taking a vector and turning it into a raster image with the node, to then be able to edit with the raster Image nodes. A image to vector node might be a good idea too though.

**Comment by AceroMagico** [comment]
I am glad you like the idea of having an importer and exporter of Svg . 
I can assure you that when you make it available many people will be happy about it.

Again as in the case of the removebg, the link to the svg editor was just a reference because it was created by the same programmer as the importer and exporter that is in auto1111 and it was not my intention to point it out to you as a "to do" ;-)

I totally agree with you that the crucial point is to have the Svg converted in raster so that you can then manipulate it with your other graphics nodes. This is exactly why I pointed you to this idea.

PlotterFun, on the other hand, is a case in point, because it allows for graphical effects that other nodes cannot currently have. Try the online demo:

https://mitxela.com/plotterfun/

Is a collection of js routines that have made history in graphical manipulation of Svg in the past years.
And they are also used by those who then make Gcode paths for Cnc machines 
(this is another aspect that can lead to using ComfyUI for this purpose as well if you will make the Svg importer and exporter).

However, is not strictly necessary that you integrate them into your nodes if you're not convinced to do it, because if you make the importer we can use PlotterFun externally to ComfyUI and then import the Svg .

**Reply by WASasquatch** [reply]
> PlotterFun, on the other hand, is a case in point, because it allows for graphical effects that other nodes cannot currently have. Try the online demo:
> 
> https://mitxela.com/plotterfun/
> 
> Is a collection of js routines that have made history in graphical manipulation of Svg in the past years. And they are also used by those who then make Gcode paths for Cnc machines (this is another aspect that can lead to using ComfyUI for this purpose as well if you will make the Svg importer and exporter).
> 
> However, is not strictly necessary that you integrate them into your nodes if you're not convinced to do it, because if you make the importer we can use PlotterFun externally to ComfyUI and then import the Svg .

I think the biggest issue is *it is JS* which I don't do (I don't really like JavaScript while I know some. To elaborate, I took many classes and years of etiquette around using JS as a tool not something to rely on in any instance, though times have changed). This is all Python. Additionally, as I've mentioned several times, ComfyUI has no native support for doing anything with nodes to send anything to the back-end except their static inputs. 

There is a script/modification that does something similar, but I'm not relying on third party plugins, especially hacked in when ComfyUI is very young and it's highly likely that route will break.

**Comment by AceroMagico** [comment]
JS doesn't drive me crazy either, there is the possibility of using comfyUI\web\extensions to inject JS into the node , just as MultiAreaConditioning.js does for example. 

But again, it is not necessary to have Plotterfun built in, if you do the Svg import and export.

**Reply by WASasquatch** [reply]
That means adding extensions which target code, that again, might change. Extensions have apparently already be breaking WAS Node Suite with breaking image data to `None` (random seed widget) in new versions of Comfy. There shouldn't need to be an extension for simply defining a nodes output to begin with.

**Comment by AceroMagico** [comment]
Of course there should be no need for it, as an excuse there is the fact as you say that the application is in its infancy, as a criticism you can say that there is not a great willingness to open it up to other developers...
But of course everyone manages their own applications as they think they should.

**Reply by WASasquatch** [reply]
>  But of course everyone manages their own applications as they think they should.

Or abandoned, and you rely on, etc. And this is a node suite, not an extension. It uses node functionality, which is what ComfyUI provides. And these are basic things ComfyUI needs. And I don't think that arguments valid for a in development software when relying on third party modifications you don't manage that may be abandoned, grow incompatible, and then your node is no longer function-able with an update that you don't have control over. That's just negligent foresight imo.

**Comment by AceroMagico** [comment]
sorry for the delay but busy times.

`But of course everyone manages their own applications as they think they should.`
was not referring to you this sentence.

Comfy is not written out of thin air it is based on Littlegraph as you know and the concept of a custom node "castrated" of basics in and out functionality is a desire not a necessity... concept completely absent in Littlegraph. 
(I know one is JS and the other Python if that's your next answer :D)

**Comment by WASasquatch** [comment]
Anyways, I have SVG converting planned for possibly today. Working on a [file] history node idea.

**Comment by AceroMagico** [comment]
Very Cool :)

**Comment by WASasquatch** [comment]
SVG will have to wait, I guess. The three options I checked all required system-level packages that I don't want WAS Suite editing. The one option I thought would work turns out to be using some sort of API which isn't working on Colab or my system complaining about being in a Async loop or something when using their example.
